### PR TITLE
tests: Added limitation test cases for V2 resources

### PR DESCRIPTION
**Pull request by lucashuy**
#### Which issue(s) does this change fix?

None

#### Why is this change necessary?
Adds the tested that is needed to validate limitation catching and handling behaviour.

#### How does it address the issue?
Adds the following limitation tests:
- testing local linking
- testing multiple resource linking
- testing OpenApi body definition

Additionally, the above test cases are also run through the `terraform apply` test class to validate the applied project can be run with `sam local start-api`.

#### What side effects does this change have?
`Body` is now being mapped for the V2 Api resource.

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [x] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by moelasmar — APPROVED** [review]

File: tests/integration/testdata/start_api/terraform/terraform-v2-api-simple/main.tf
Comment (moelasmar) [inline tests/integration/testdata/start_api/terraform/terraform-v2-api-simple/main.tf:54]: for resource `"aws_apigatewayv2_route" "example"` .. I think the `target` property value should be in format `"integrations/${<integration id>}"`. If it is working using this format when we run `terraform apply`, please add another example to use the previous format as explained in [the doc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apigatewayv2_route)

**Review by mildaniel — COMMENTED** [review]
LGTM. I'll defer to Mohamed's comment and approve once that's in

**Review by moelasmar — APPROVED** [review]

**Review by mildaniel — APPROVED** [review]

**Pull request merged**
